merged-tests branch - testing contribution

wagscoll opened the pull request:
This testing branch was created as part of a project for CS4499 - Unit Testing. 

Contributors: Collin (wagscoll), Alex (Svanalex). 


- Implemented comprehensive exception handling in `managers/utils/validation.py` using two custom exceptions:
  - InvalidFormDataException
  - AutocompleteOptionMismatchException
  
- Added test files for 5 modules/components:
  - tests/utils/test_validation.py
  - tests/test_config.py
  - tests/components/modules/test_incomemode.py
  - tests/managers/test_person.py
  - tests/modules/test_people.py
  
- Updated `pyproject.toml` to configure pytest and include all test files.
- Added TESTING.md documenting how to run tests, exceptions, and coverage summary.

Modified:
- src/bagels/managers/utils/validation.py
- pyproject.toml

Created:
- tests/utils/test_validation.py
- tests/test_config.py
- tests/components/modules/test_incomemode.py
- tests/managers/test_person.py
- tests/modules/test_people.py
- TESTING.md

Notable coverage:
- components/modules/incomemode.py: 96%
- components/modules/people.py: 87%
- utils/validation.py: 84%
- config.py: 69%
- managers/persons.py: 69%
- Total statements: 3,375
- Statements covered: 1,030
- Overall coverage: 31%

More information can be found within: TESTING.md

svanalex committed:
fix: add tests for record modals

Commit:
Add validateForm unit tests for number, date, dateAutoDay, and autocomplete fields, and pyproject.toml dependencies update to include pytest

Commit:
Implemented Exception Handling in validation.py & optimized test_validation.py to handle exceptions.

Commit:
Added tests for components/modules/test_incomemode.py

Commit:
Added coverage for config.py

svanalex committed:
fix: updated test cases in test_person.py, added framework for testing record.py

Commit:
Updated pyproject.toml to include dependencies AND filtered tests to only run the one's written for the purpose of CS4499.

Commit:
Removing test_record & test_record_alt from pyproject.toml.

Commit:
Added TESTING.md.

svanalex committed:
feat: added tests for people.py

svanalex committed:
feat: add tests for account mode and exception handling

Commit:
Adding uv.lock

Commit:
Merge branch 'add-tests-and-exceptions' into merged-tests
Merging both testing branches.

wagscoll committed:
Update TESTING.md

wagscoll committed:
Rename test file from test_people.py to test_person.py

wagscoll committed:
Enhance testing guide and validation error handling

Updated the Bagels Testing Guide with improved instructions, modified files, and detailed exception handling for validation errors.

Commit:
Added test_people.py to pyproject.toml > testpaths

EnhancedJax commented:
Thank you for contributing to bagels! Will review this PR soon. Best of luck to your course project!

wagscoll committed:
Test workflow (#15)

* GH Workflow addition

* Update coverage action to use VeryGoodOpenSource

* Update GitHub Actions workflow for test coverage

* Add step descriptions to GitHub Actions workflow

* Limit test coverage to src/bagels directory

* Update test workflow to include specific coverage files

* Refactor test workflow for improved coverage reporting

Updated the test workflow to run all tests with coverage and added a summary step.

* Refactor test workflow for improved coverage reporting

Updated test workflow to run tests with coverage for the entire bagels module and added coverage percentage output.

* Change minimum coverage threshold from 80% to 30%

* Update comments in GitHub Actions workflow

* Update GitHub Actions workflow for testing coverage

* Refactor GitHub Actions workflow for testing

* Refine coverage options in test workflow

Updated coverage reporting to include specific modules.

* Change minimum coverage requirement to 30%

Updated minimum coverage threshold from 80% to 30% in CI workflow.

* Change minimum coverage requirement to 30%

---------

Co-authored-by: wagscoll <collin@DESKTOP-BJIE6R2>
